src/drawer.multicanvas.js

```diff
@@ -52,28 +52,30 @@ WaveSurfer.util.extend(WaveSurfer.Drawer.MultiCanvas, {
             this.removeCanvas();
         }
 
+        var leftOffset = 0;
         for (var i in this.canvases) {
             // Add some overlap to prevent vertical white stripes, keep the width even for simplicity.
-            var canvasWidth = this.maxCanvasWidth + 2 * Math.ceil(this.params.pixelRatio / 2);
+            var canvasWidth = Math.floor(totalWidth / requiredCanvases * this.params.pixelRatio),
+                canvasOverlap = 2 * Math.ceil(this.params.pixelRatio / 2),
+                actualCanvasWidth = canvasWidth;
 
-            if (i == this.canvases.length - 1) {
-                canvasWidth = this.width - (this.maxCanvasWidth * (this.canvases.length - 1));
+            if (i < this.canvases.length - 1) {
+                actualCanvasWidth += canvasOverlap;
             }
-
-            this.updateDimensions(this.canvases[i], canvasWidth, this.height);
+ 
+            this.updateDimensions(this.canvases[i], actualCanvasWidth, this.height, leftOffset);
             this.clearWaveForEntry(this.canvases[i]);
+            leftOffset += canvasWidth / this.params.pixelRatio;
         }
     },
 
      addCanvas: function () {
         var entry = {};
-        var leftOffset = this.maxCanvasElementWidth * this.canvases.length;
 
         entry.wave = this.wrapper.appendChild(
             this.style(document.createElement('canvas'), {
                 position: 'absolute',
                 zIndex: 1,
-                left: leftOffset + 'px',
```

Comment:
I now calculate the left offset in `updateDimensions`. This is because the left offset will change as the canvases resize, whereas only the last canvas used to change size, and all of the left offsets were constant.